src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LoweredDynamicOperationFactory.cs

```diff
@@ -98,7 +98,7 @@ private enum CSharpArgumentInfoFlags
                 _factory.Typeof(resultType),
 
                 // context:
-                _factory.Typeof(_factory.CurrentType)
+                _factory.Typeof(_factory.CompilationState.DynamicOperationContextType)
```

Comment:
Consider adding a DynamicOperationContextType property to SyntheticBoundNodeFactory.